src/msssync.py

```diff
@@ -5,53 +5,154 @@
 
-def facility_sync(syncserver, syncfacilityid):
+def get_sync_config_file_location(sync_config_filename):
+    kolibri_home = os.environ.get("KOLIBRI_HOME")
+    syncini_file = os.path.join(kolibri_home, sync_config_filename)
+    return syncini_file
+
+def fetch_remote_sync_file(sync_config_filename, facility_id):
+    try:
+        url = 'http://content.myscoolserver.in/configs/' + facility_id + '/' + sync_config_filename
+        r = requests.get(url, allow_redirects=True)
+        syncini_file = get_sync_config_file_location(sync_config_filename)
+        open(syncini_file, 'wb').write(r.content)
+        return syncini_file
+    except RequestException:
+        # Need to inform the user to connect the device to the Internet
+        return fetch_remote_sync_file(sync_config_filename, facility_id)
+
+def update_sync_config_file(sync_config_filename, facility_id):
+    syncini_file = fetch_remote_sync_file(sync_config_filename, facility_id)
+    delete_import_credentials(syncini_file)
+
+def get_sync_params(syncini_file, grade):
+    configur = ConfigParser()
+
+    try:
+        file = open(syncini_file, 'r')
+    except IOError:
+        logging.info("Facility sync file not available")
+
+    configur.read(syncini_file)
+    sync_params = {}
+    sync_params['sync_on'] = configur.getboolean('DEFAULT', 'SYNC_ON')
+    sync_params['sync_delay'] = configur.getfloat('DEFAULT', 'SYNC_DELAY')
+    sync_params['sync_server'] = configur.get('DEFAULT', 'SYNC_SERVER')
+    sync_params['channel'] = configur.get('DEFAULT','CHANNEL')
+    sync_params['node'] = configur.get('DEFAULT', grade + '_NODE')
+    sync_params['sync_user'] = configur.get('DEFAULT', 'SYNC_ADMIN', fallback=None)
+    sync_params['sync_password'] = configur.get('DEFAULT', 'SYNC_ADMIN_PASSWORD', fallback=None)
+
+    return sync_params
+
+def delete_import_credentials(syncini_file):
+    configur = ConfigParser()
+
+    try:
+        file = open(syncini_file, 'r')
+        configur.read(syncini_file)
+        configur.remove_option('DEFAULT', 'SYNC_ADMIN')
+        configur.remove_option('DEFAULT', 'SYNC_ADMIN_PASSWORD')
+        with open(syncini_file,"w") as configfile:
+            configur.write(configfile)
+    except IOError:
+        logging.info("Facility sync file not available")
+
+def import_facility(sync_params, facility_id):
     pid = os.fork()
     if pid == 0:
-        main(["manage", "sync", "--baseurl", syncserver, "--facility", syncfacilityid, "--verbosity", "3"])
+        main(["manage", "sync", "--baseurl", sync_params['sync_server'], "--facility", facility_id, "--verbosity", "3", "--username", sync_params['sync_user'], "--password", sync_params['sync_password'], "--no-push", "--noninteractive"])
     else:
         os.waitpid(pid, 0)
 
+def import_channel(channel_id):
+    pid = os.fork()
+    if pid == 0:
+        main(["manage", "importchannel", "network", channel_id])
+    else:
+        os.waitpid(pid, 0)
+
+def import_content(channel_id, content_node):
```

Comment:
https://github.com/learningequality/kolibri/pull/7431

Find out how to get list of all resources in a lesson and lesson status, then we can download only activated lessons and remove once lesson is complete

There must be a way to get list of resources, so instead of "Resource not found on this device", we can get list of resources not found and use that instead of manually listing content nodes in config file.

Reply:
As per discussion with teachers as yet, they would want lessons of entire year on the device, if not at the beginning of the year then by the end. This seems logical for referral and revision purposes.

Hence for trials we are incrementally moving content linked to active lesson plans into a "released" top content node corresponding to a grade of the facility channel.

Since the auto content sync corresponding to lesson plans is a planned feature of Kolibri, will not delve into developing this until there is an absolute need while we move further ahead with the trials. 

Need to first and foremost validate if mobile or Android device based learning is at all a practical model for our solution adopters.